docs(weekly): add this week in databend 107 (#12552)
datafuselabs/databend: website/blog/2023-08-20-databend-weekly-107.md

## What's On In Databend

Stay connected with the latest news about Databend.

### Understanding Connection Parameters

The connection parameters refer to a set of essential connection details required for establishing a secure link to supported external storage services, like Amazon S3. These parameters are enclosed within parentheses and consists of key-value pairs separated by commas or spaces. It is commonly utilized in operations such as creating a stage, copying data into Databend, and querying staged files from external sources. 

For example, the following statement creates an external stage on Amazon S3 with the connection parameters:

```sql
CREATE STAGE my_s3_stage
URL = 's3://load/files/'
CONNECTION = (
    ACCESS_KEY_ID = '<your-access-key-id>',
    SECRET_ACCESS_KEY = '<your-secret-access-key>'
);
```

If you are interested in learning more, please check out the resources listed below.

- [Docs | SQL Reference - Connection Parameters](https://databend.rs/doc/sql-reference/connect-parameters)

### Adding Storage Parameters for Hive Catalog

> Over the past week, Databend has introduced storage parameter for Hive Catalog, which allows configuring specific storage services for it, no longer relying on the storage backend of Default Catalog itself.

Reviewer: ```suggestion
Over the past week, Databend introduced storage parameters for the Hive Catalog, allowing the configuration of specific storage services. This means that the catalog no longer relies on the storage backend of the default catalog.
```

> The following example shows how to create a Hive Catalog with MinIO as the underlying storage service:

Reviewer: ```suggestion
The following example shows how to create a Hive Catalog using MinIO as the underlying storage service:
```

## Code Corner

Discover some fascinating code snippets or projects that showcase our work or learning journey.

### Using `gitoxide` to Speed Up Git Dependency Downloads

`gitoxide` is a high-performance, modern Git implementation written in Rust. Utilizing the `gitoxide` feature of cargo (Unstable), the `gitoxide` crate can replace `git2` to perform various Git operations, thereby achieving several times performance improvement when downloading crates-index and git dependencies.

Databend has recently enabled this feature for `cargo {build | clippy | test}`` in CI. You can also try to add the -Zgitoxide option to speed up the build process during local development:

```bash
cargo -Zgitoxide=fetch,shallow-index,shallow-deps build
```

If you are interested in learning more, please check out the resources listed below:

- [chore(ci): cargo build with -Zgitoxide](https://github.com/datafuselabs/databend/pull/12504)
- [The Cargo Book | Unstable - gitoxide](https://doc.rust-lang.org/nightly/cargo/reference/unstable.html#gitoxide)

## Highlights

We have also made these improvements to Databend that we hope you will find helpful:

> - `VALUES` clause can be used without being combined with `SELECT`.

Reviewer: Doc: https://databend.rs/doc/sql-commands/query-syntax/values

> - Setting default values is supported when modifying column type.

Reviewer: You can now set a default value when modifying the type of a column. See https://databend.rs/doc/sql-commands/ddl/table/alter-table-column for details.

> - **Virtual Column** support has been added for tables in Parquet format.

Reviewer: ```suggestion

```

> - Automatic reclustering of tables is now supported after performing write operations (`COPY INTO` and `REPLACE INTO`).

Reviewer: Databend now automatically recluster a table after a write operation such as COPY INTO and REPLACE INTO.

## What's Up Next

We're always open to cutting-edge technologies and innovative ideas. You're more than welcome to join the community and bring them to Databend.

> ### Enhancing `infer_schema` Functionality to Support All File Locations

Reviewer: ```suggestion
### Enhancing `infer_schema` for All File Locations
```

> Currently, it is possible to query files using file locations or stages in our database. Examples of this are:

Reviewer: ```suggestion
Currently, it is possible to query files using file locations or from stages in Databend.
```

```sql
select * from 'fs:///home/...';
select * from 's3://bucket/...';
select * from @stage;
```

> However, when it comes to invoking the `infer_schema` function, it is limited to only staged files. For example:

Reviewer: ```suggestion
However, the `infer_schema` function only works with staged files. For example:
```

```sql
select * from infer_schema(location=>'@stage/...');
```

> When trying to use `infer_schema` with other file locations, it causes a panic:

Reviewer: When attempting to use infer_schema with other file locations, it leads to a panic:

```sql
select * from infer_schema(location =>'fs:///home/...'); -- this will panic.
```

> Therefore, the required enhancement is to expand the `infer_schema` functionality to allow inference on all types of file paths, not just those from staged files. This change will improve the consistency of our system and make the `infer_schema` function more useful.

Reviewer: ```suggestion
So, the improvement involves extending the infer_schema capability to encompass all types of file paths, not limited to staged files. This will enhance system consistency and the usefulness of the infer_schema function.```
